lib/_http_client.js

```diff
@@ -171,14 +171,14 @@ function ClientRequest(options, cb) {
     }
     if (host && !this.getHeader('host') && setHost) {
       var hostHeader = host;
-      var posColon = -1;
 
       // For the Host header, ensure that IPv6 addresses are enclosed
       // in square brackets, as defined by URI formatting
       // https://tools.ietf.org/html/rfc3986#section-3.2.2
-      if (-1 !== (posColon = hostHeader.indexOf(':')) &&
-          -1 !== (posColon = hostHeader.indexOf(':', posColon)) &&
-          '[' !== hostHeader[0]) {
+      var posColon = hostHeader.indexOf(':');
```

Comment:
I was sticking to the "style" of the original code. I'm also still reluctant to use `const` in runtime code because I've been burned so many times by it in the past (even when it hasn't triggered a permanent deopt).